client/src/test/java/com/cyberark/conjur/sdk/endpoint/AuthenticationExternalApiTest.java

```diff
@@ -0,0 +1,409 @@
+package com.cyberark.conjur.sdk.endpoint;
+
+import com.cyberark.conjur.sdk.*;
+import com.cyberark.conjur.sdk.auth.*;
+import com.cyberark.conjur.sdk.model.ServiceAuthenticators;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+/**
+ * API tests for AuthenticationApi.
+ */
+public class AuthenticationExternalApiTest extends ConfiguredTest {
+
+    private AuthenticationApi api;
+    private AuthenticationApi badAuthApi;
+    private AuthenticationApi aliceApi;
+    private AuthenticationApi mockApi;
+    private ApiClient mockClient;
+    private ArgumentCaptor<Map> mapCaptor;
+
+    /**
+     * Set up Authentication API clients to test.
+     * 
+
+     * @throws ApiException if it fails to setup the necessary clients/APIs
+     */
+    @Before
+    public void setUpClients() throws ApiException {
+        api = new AuthenticationApi();
+        badAuthApi = new AuthenticationApi(nonAuthClient);
+
+        // Setup Alice API
+        ApiClient aliceClient = getApiClient("alice");
+        aliceClient.setPassword("alice");
+        aliceApi = new AuthenticationApi(aliceClient);
+
+        // Setup mock API + response
+        mockClient = Mockito.mock(ApiClient.class);
+        mockApi = new AuthenticationApi(mockClient);
+        ApiResponse<Object> mockResponse = new ApiResponse<Object>(200, null, "Success!");
+
+        Mockito.when(mockClient.escapeString(Mockito.anyString())).thenCallRealMethod();
+        Mockito.when(mockClient.parameterToString(Mockito.anyString())).thenCallRealMethod();
+        Mockito.when(mockClient.selectHeaderAccept(Mockito.any(String[].class)))
+            .thenCallRealMethod();
+        Mockito.when(mockClient.selectHeaderContentType(Mockito.any(String[].class)))
+            .thenCallRealMethod();
+        Mockito.when(mockClient.execute((okhttp3.Call) Mockito.any(), (Type) Mockito.any()))
+            .thenReturn(mockResponse);
+
+        // Create a mechanism to capture map param values in mock API calls
+        mapCaptor = ArgumentCaptor.forClass(Map.class);
+    }
+
+    /**
+     * Get the API key admin user's API key from an env var.
+     *
+     * @return API key
+     */
+    public String apiKey() {
+        return System.getenv("CONJUR_AUTHN_API_KEY");
+    }
+
+    /**
+     * Get the access token via an OIDC call to keycloak.
+     *
+     * @return OIDC Access token
+     * @throws IOException when the API call fails
+     */
+    public String getOidcIdToken() throws IOException {
```

Comment:
Wondering if it would be a better idea to have this function live in `ConfiguredTest.java` next to `setupOidcWebservice`. Just a preference thing, though, feel free to ignore as it probably isn't used anywhere else.